src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -429,6 +434,123 @@ GetShardCost(uint64 shardId, void *voidContext)
 }
 
 
+/*
+ * worker_partitioned_relation_total_size returns the total size (including indexes) of
+ * a partitioned relation with its partitions, using the function pg_partition_tree.
+ */
+Datum
+worker_partitioned_relation_total_size(PG_FUNCTION_ARGS)
+{
+	char *relname = text_to_cstring(PG_GETARG_TEXT_P(0));
+
+	/* pg_partition_tree finds the partitions recursively and returns them */
+	char *queryText = "SELECT sum(pg_total_relation_size(relid))::bigint AS total_size "
```

Comment:
Given that we cast the sum of `pg_total_relation_size` to a `bigint`, I expect the return type of `pg_total_relation_size` is of a different type, possibly a smaller width integer. What would happen if the sum overflows the width of that type? Will we still get the correct answer?

What is the reason we cast the result of the sum and not before summing up the values?

Reply:
https://www.postgresql.org/docs/9.4/functions-admin.html#FUNCTIONS-ADMIN-DBSIZE

All size functions return `bigint` actually. I missed that out.